xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/internal/migration/AbstractDocumentsMigration.java

```diff
@@ -63,28 +58,13 @@ public abstract class AbstractDocumentsMigration extends AbstractHibernateDataMi
     @Override
     protected void hibernateMigrate() throws DataMigrationException
     {
-        List<XWikiDocument> documents = selectDocuments()
-            .stream()
-            .map(this::convert)
-            .filter(Optional::isPresent)
-            .map(Optional::get)
-            .collect(toList());
-        logBeforeQueuingTasks(documents);
-        for (XWikiDocument document : documents) {
-            logBeforeQueuingTask(document);
-            this.taskManager.addTask(this.getXWikiContext().getWikiId(), document.getId(), getTaskType());
-        }
-    }
+        List<String> selectedDocuments = selectDocuments();
+        logBeforeQueuingTasks(selectedDocuments);
 
-    private Optional<XWikiDocument> convert(String documentReference)
-    {
-        XWikiContext context = getXWikiContext();
-        try {
-            return Optional.of(
-                context.getWiki().getDocument(this.documentReferenceResolver.resolve(documentReference), context));
-        } catch (XWikiException e) {
-            this.logger.error("Failed to resolve [{}]", documentReference, e);
-            return Optional.empty();
+        for (String documentReference : selectedDocuments) {
+            XWikiDocument document = new XWikiDocument(this.documentReferenceResolver.resolve(documentReference));
```

Comment:
Hmm I did not notice that in your previous version but `this.documentReferenceResolver` is the default `DocumentReferenceResolver`, which means that the missing wiki part is always going to be resolved as "xwiki" so this code does not really work in subwikis.

Reply:
Even when the wiki in the context is a subwiki?

Reply:
> Even when the wiki in the context is a subwiki?

The default resolver does not care about that, you need the "current" resolver to take the context into account.

Reply:
See https://github.com/xwiki/xwiki-platform/pull/2241 for the bugfix of that part.